src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -123,6 +123,7 @@ pub trait Linker {
     fn pgo_gen(&mut self);
     fn control_flow_guard(&mut self);
     fn debuginfo(&mut self);
```

Comment:
This needs to be merged into `strip`, for MSVC in particular.

(Also, looks like for PTX linker the effect of strip can be achieved by *not* passing `--debug` like `fn debuginfo` durrently does.)

Reply:
Actually, it's probably better to keep the name `debuginfo` after merging.
```
fn debuginfo(&mut self, strip: Strip);
```

Reply:
Do we need to keep the `opts.debuginfo` here? https://github.com/rust-lang/rust/blob/7c59a81a5fcbaaca311f744cd7c68d99bfbb05d3/src/librustc_codegen_ssa/back/linker.rs#L892-L898

Reply:
Maybe? I don't know anything about Emscripten.

Reply:
I am inclined to keep the function `debuginfo` as it is. There are also https://github.com/rust-lang/rust/blob/7c59a81a5fcbaaca311f744cd7c68d99bfbb05d3/src/librustc_codegen_ssa/back/linker.rs#L1187-L1189 and https://github.com/rust-lang/rust/blob/7c59a81a5fcbaaca311f744cd7c68d99bfbb05d3/src/librustc_codegen_ssa/back/linker.rs#L689-L713

Reply:
Both of these need merging because `--debug`, `/DEBUG`, and `/NATVIS` only need to be passed in the `Strip::None` case.

Reply:
Thanks. I merged them. I added https://github.com/rust-lang/rust/blob/7c59a81a5fcbaaca311f744cd7c68d99bfbb05d3/src/librustc_codegen_ssa/back/linker.rs#L892-L898 to the `finalize` function.

Reply:
It's better to keep them in their old place, the options clearly preserve or not preserve some debuginfo, so they should work together with stripping.

Also, I suspect (although not sure) that `--llvm-opts` is an `emcc` frontend's options so the linker won't understand them.
Again, it's not necessary to guess this now, it's always an option to keep the status quo behavior. If there are people interested in the `-Z strip` support for emscripten, they will contribute it later.

(Also, nobody knows what `fn finalize` means, I'll try to remove it some time later.)